Balance command

jcarpanelli opened the pull request:
Fixes #828, fixes #829.

Started implementing both interactive command and script to get the balance of an account on a specified network, or to get the balance in tokens if an ERC20 address is specified.

Example usage:
![balance-gif](https://i.ibb.co/gv0Nq88/balance.gif)

Still have to add some tests, and also decide what to do with the ERC20 `abi` that I included in a new `utils/constants.ts`, which I believe it's mandatory to have in the project to query both token balance and symbol.

~_Message for the reviewer: I branched from `feature/transfer-command` and squashed all the commits implemented on it for making the review easier. You should review all commits after `f3f0b76` (included). After merging https://github.com/zeppelinos/zos/pull/823, we can remove that squashed commit_.~

spalladino commented on packages/lib/src/artifacts/ZWeb3.ts:
```diff
@@ -50,6 +50,10 @@ export default class ZWeb3 {
     return Web3.utils.toWei(value, unit);
   }
 
+  public static fromWei(value: string, unit: any): string {
```
I'd consider removing this and other utils functions from here. They are in zweb3 only because they are implemented by web3, but they are actually unrelated to the connection to the network. We can tackle this later, of course.

spalladino commented on packages/cli/src/commands/balance.ts:
```diff
@@ -0,0 +1,40 @@
+  .command(signature, undefined, { noHelp: true })
+  .usage('[options]')
+  .description(description)
+  .option('--erc20 <contractAddress>', 'specify an ERC20 token address')
```
I'd add something along the lines of "or leave empty to query ETH balance"

jcarpanelli commented:
Ready for another review :racehorse:. I implemented the suggestions above, added tests for `units.ts`, and for the balance command and script.
I also used `bignumber.js` (which is already installed in the `cli` package) to do some simple operations, but as we have been discussing, we should consider moving to `BN.js`, which is used by `web3@1.0.0`.

spalladino commented on packages/cli/src/commands/balance.ts:
```diff
@@ -0,0 +1,43 @@
+  .command(signature, undefined, { noHelp: true })
+  .usage('[options]')
+  .description(description)
+  .option('--erc20 <contractAddress>', 'specify an ERC20 token address or leave empty to query ETH balance')
```
```suggestion
  .option('--erc20 <contractAddress>', 'query the balance of an ERC20 token instead of ETH')
```
"Leave empty" seems to indicate that the value of the flag should be left empty, not that the user should not set the flag at all.

spalladino commented on packages/cli/src/commands/balance.ts:
```diff
@@ -0,0 +1,43 @@
+    ...networksList('network', 'list'),
+    accountAddress: {
+      type: 'input',
+      message: 'Enter an account address',
```
```suggestion
      message: 'Enter an address to query its balance',
```

spalladino commented on packages/cli/src/models/network/TransactionController.ts:
```diff
@@ -1,23 +1,57 @@
+      const contract = ZWeb3.contract(ERC20_PARTIAL_ABI, contractAddress);
+      balance = await contract.methods.balanceOf(accountAddress).call();
+      tokenSymbol = await contract.methods.symbol().call();
+      tokenDecimals = await contract.methods.decimals().call();
```
A `Promise.all` to query all simultaneously is usually better in these scenarios, as it cuts down the latency time to 1/3rd.

spalladino commented on packages/cli/src/utils/units.ts:
```diff
@@ -1,5 +1,22 @@
+
+export function prettifyTokenAmount(amount: string, decimals?: string, symbol?: string): string {
+  const prettifiedAmount = decimals
+    ? new BN(amount).dividedBy(new BN(10).exponentiatedBy(decimals)).toFormat()
```
```suggestion
    ? new BN(amount).shiftedBy(-decimals).toFormat()
```
See http://mikemcl.github.io/bignumber.js/#shift
